src/frontend/apps/lti_site/apps/bbb/components/UploadDocuments/index.tsx

```diff
@@ -0,0 +1,253 @@
+import { Anchor, Box, Button, Paragraph, Text } from 'grommet';
+import Dropzone from 'react-dropzone';
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { PlusSVG, ValidSVG } from 'lib-components';
+import { UploadableObjectProgress } from 'components/UploadableObjectProgress';
+import {
+  UploadManagerStatus,
+  useUploadManager,
+} from 'components/UploadManager';
+
+import { bbbAppData } from 'apps/bbb/data/bbbAppData';
+import {
+  useClassroomDocuments,
+  useUpdateClassroomDocument,
+} from 'apps/bbb/data/queries';
+import { createClassroomDocument } from 'apps/bbb/data/sideEffects/createClassroomDocument';
+import { modelName } from 'apps/bbb/types/models';
+import { truncateFilename } from 'apps/deposit/utils/truncateFilename';
+
+const messages = {
+  dropzonePlaceholder: {
+    defaultMessage: "Drag 'n' drop some files here, or click to select files",
+    description: 'Placeholder for dropzone.',
+    id: 'apps.deposit.components.DashboardStudent.UploadFiles.dropzonePlaceholder',
+  },
+  uploadButtonLabel: {
+    defaultMessage: 'Upload',
+    description: 'Label for upload button.',
+    id: 'apps.deposit.components.DashboardStudent.UploadFiles.uploadButtonLabel',
+  },
+  uploadingFile: {
+    defaultMessage:
+      'Upload in progress... Please do not close or reload this page.',
+    description:
+      'Help text to warn user not to navigate away during file upload.',
+    id: 'apps.deposit.components.DashboardStudent.UploadFiles.uploadingFile',
+  },
+  isDefaultDocument: {
+    defaultMessage: 'Default document',
+    description: 'Helper message for default document.',
+    id: 'apps.deposit.components.DashboardStudent.UploadFiles.isDefaultDocument',
+  },
+  setDefaultDocument: {
+    defaultMessage: 'Click to set as default document',
+    description: 'Button helper message for setting a document as default.',
+    id: 'apps.deposit.components.DashboardStudent.UploadFiles.setDefaultDocument',
+  },
+  downloadButtonLabel: {
+    defaultMessage: 'Download',
+    description: 'Label for download button.',
+    id: 'apps.deposit.components.DashboardStudent.UploadFiles.downloadButtonLabel',
+  },
+};
+
+type UploadDocumentsRowProps = {
+  filename: string;
+  isDefault?: boolean;
+  documentId?: string;
+  uploadingObjectId?: string;
+  uploadState?: string;
+  url?: string;
+};
+
+const UploadDocumentsRow = ({
+  filename,
+  isDefault,
+  documentId,
+  uploadingObjectId,
+  uploadState,
+  url,
+}: UploadDocumentsRowProps) => {
+  const intl = useIntl();
+
+  let setDefaultDocument = () => {};
+  if (uploadState === 'ready' && documentId) {
+    const updateClassroomMutation = useUpdateClassroomDocument(documentId);
```

Comment:
You should not called conditional hook, this is susceptible to crash (https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)

Reply:
How would you manage that ?
(I get no error with this code, btw)